homeassistant/components/alexa/smart_home.py

```diff
@@ -911,8 +947,11 @@ class _Cause:
 class Config:
     """Hold the configuration for Alexa."""
 
-    def __init__(self, should_expose, entity_config=None):
+    def __init__(self, client_id, client_secret, should_expose,
+                 entity_config=None):
         """Initialize the configuration."""
+        self.client_id = client_id
```

Comment:
We shouldn't do it like this, instead we should pass in an awaitable method to get a valid access token.

Client ID/Secret is only an implementation detail for the locally hosted one and not for Home Assistant Cloud.

Reply:
I wanted to leave that to a new PR. But I can add it to this one if you prefer.

Reply:
We should do it right to avoid a breaking change.